controllers/accounts_test.go

```diff
@@ -259,6 +270,203 @@ func (suite *AccountsTestSuite) TestAccountDeletionEndpointEnabled() {
+	suite.Len(*finalizeParsedResp.RecoveryKey, 32)
+	suite.NotEmpty(*finalizeParsedResp.RecoveryKey)
+
+	_, err = base32.StdEncoding.DecodeString(*finalizeParsedResp.RecoveryKey)
+	suite.Require().NoError(err)
+
+	// Verify 2FA is now enabled
+	updatedAccount, err := suite.ds.GetAccount(nil, account.Email)
+	suite.Require().NoError(err)
+	suite.True(updatedAccount.IsTwoFAEnabled())
+	suite.NotNil(updatedAccount.RecoveryKeyHash)
+	suite.True(util.VerifyRecoveryKeyHash(*finalizeParsedResp.RecoveryKey, updatedAccount.RecoveryKeyHash))
+
+	// Test initializing TOTP when it's already enabled
+	initReq = util.CreateJSONTestRequest("/v2/accounts/2fa/totp/init", controllers.TwoFAInitRequest{
+		GenerateQR: true,
+	})
+	initReq.Header.Set("Authorization", "Bearer "+token)
+	initResp = util.ExecuteTestRequest(initReq, suite.router)
+	suite.Equal(http.StatusBadRequest, initResp.Code)
+	util.AssertErrorResponseCode(suite.T(), initResp, util.ErrTOTPAlreadyEnabled.Code)
+}
+
+func (suite *AccountsTestSuite) TestDisableTOTP() {
+	token, account := suite.createAuthSession()
+
+	// Enable 2FA first
+	err := suite.ds.SetTOTPSetting(account.ID, true)
+	suite.Require().NoError(err)
+
+	// Generate and store a recovery key
+	recoveryKey := "MFZWIZTBONSWM53BONSWMYLTMVTGC43F"
+	err = suite.ds.SetRecoveryKey(account.ID, &recoveryKey)
+	suite.Require().NoError(err)
+
+	// Verify timestamps are set
+	details, err := suite.ds.GetTwoFADetails(account.ID)
+	suite.Require().NoError(err)
+	suite.NotNil(details.TOTPEnabledAt)
+	suite.NotNil(details.RecoveryKeyCreatedAt)
+
+	// Test disabling TOTP
+	disableReq := httptest.NewRequest("DELETE", "/v2/accounts/2fa/totp", nil)
+	disableReq.Header.Set("Authorization", "Bearer "+token)
+	disableResp := util.ExecuteTestRequest(disableReq, suite.router)
+	suite.Equal(http.StatusNoContent, disableResp.Code)
+
+	// Verify 2FA is now disabled
+	updatedAccount, err := suite.ds.GetAccount(nil, account.Email)
+	suite.Require().NoError(err)
+	suite.False(updatedAccount.IsTwoFAEnabled())
+	suite.Nil(updatedAccount.RecoveryKeyHash)
+
+	// Verify timestamps are cleared
+	details, err = suite.ds.GetTwoFADetails(account.ID)
+	suite.Require().NoError(err)
+	suite.Nil(details.TOTPEnabledAt)
+	suite.Nil(details.RecoveryKeyCreatedAt)
+}
+
+func (suite *AccountsTestSuite) TestRecoveryKeyEndpoints() {
+	token, account := suite.createAuthSession()
+
+	// Test regenerate recovery key
+	req := util.CreateJSONTestRequest("/v2/accounts/2fa/recovery", nil)
+	req.Header.Set("Authorization", "Bearer "+token)
+
+	resp := util.ExecuteTestRequest(req, suite.router)
+	suite.Equal(http.StatusOK, resp.Code)
+
+	// Verify response contains a recovery key
+	var keyResp controllers.RecoveryKeyResponse
+	util.DecodeJSONTestResponse(suite.T(), resp.Body, &keyResp)
+	suite.NotEmpty(keyResp.RecoveryKey)
+
+	// Verify recovery key was stored
+	hasKey, err := suite.ds.HasRecoveryKey(account.ID)
+	suite.Require().NoError(err)
+	suite.True(hasKey)
+
+	// Verify timestamp was set
```

Comment:
One more test case that could be useful here:
1. Generate a recovery key and check a few things (i.e. what we already have)
2. Re-generate the recovery key
3. Check that the `CreatedAt` date is different (greater) than the original one from #1.
4. Delete the recovery key and check a few things (i.e. what we already have)